### PR TITLE
ACM-6842: SDK changes for creating a cluster with hs2 on master (is_hs2) 

### PR DESCRIPTION
**Pull request by surajg-q**
Provide functionality to create/update hive clusters with HS2 on master.

### CODE REVIEW

**Comment by chattarajoy** [comment]
Test cases?

**Comment by surajg-q** [comment]
@chattarajoy Test case added

File: tests/test_clusterv22.py [hunk 1]
```diff
@@ -211,3 +211,22 @@ def test_image_version_v22(self):
...
+        print(Connection._api_call)
+        Connection._api_call.assert_called_with('POST', 'clusters',
+                                                {'engine_config': {'hive_settings': {'is_hs2': 'true', 'hs2_thrift_port': '10001'}},
+                                                 'cluster_info': {'label': ['test_label'],
+                                                                  'composition': {'master': {'nodes': [{'percentage': 100, 'type': 'ondemand'}]},
+                                                                                  'min_nodes': {'nodes': [{'percentage': 100, 'type': 'ondemand'}]},
+                                                                                  'autoscaling_nodes': {'nodes': [{'percentage': 50, 'type': 'ondemand'},
+                                                                                                                  {'percentage': 50, 'type': 'spot', 'maximum_bid_price_percentage': 100, 'timeout_for_request': 1, 'allocation_strategy': None, 'fallback': 'ondemand'}]}}}})
+
+if __name__ == '__main__':
```
Comment (chattarajoy) [inline tests/test_clusterv22.py:231]: @surajg-q why is this needed?
